boa_parser/src/parser/expression/update.rs

```diff
@@ -72,11 +72,28 @@ where
                 cursor
                     .next(interner)?
                     .expect("Punctuator::Inc token disappeared");
-
+                
                 let target = UnaryExpression::new(self.name, self.allow_yield, self.allow_await)
                     .parse(cursor, interner)?;
-
-                if cursor.strict_mode() {
+                let strict = cursor.strict_mode();
+                // https://tc39.es/ecma262/#sec-update-expressions-static-semantics-early-errors
+                let ok = match &target {
+                    Expression::Identifier(_) if !strict => true,
+                    Expression::Identifier(ident)
+                        if ![Sym::EVAL, Sym::ARGUMENTS].contains(&ident.sym()) =>
+                    {
+                        true
+                    }
```

Comment:
I'm pretty sure the eval/arguments check is already done below. Can we move the check above to the `match` in order to skip rechecking this? Maybe something like:

```Rust
Expression::Identifier(ident) =>
                    {
                        check_strict_arguments_or_eval(ident, position)?;
                        true
                    }
```

Reply:
Sure, I've moved the check inside `match` for both UnaryExpression and LeftHandSideExpression to make it uniform.